src/Components/Common/MarkdownEditor.tsx

```diff
@@ -0,0 +1,396 @@
+      blockquote.parentNode.insertBefore(br, blockquote.nextSibling);
+    }
+
+    saveState();
+  };
+
+  const handleLink = () => {
+    const userLink = prompt("Enter a URL");
+    if (userLink) {
+      const selection = window.getSelection();
+      if (!selection || !selection.rangeCount) return;
+
+      const range = selection.getRangeAt(0);
+      const anchor = document.createElement("a");
+      anchor.href = userLink.startsWith("http")
+        ? userLink
+        : `http://${userLink}`;
+      anchor.textContent = range.toString();
+
+      range.deleteContents();
+      range.insertNode(anchor);
+      saveState();
+    }
+  };
+
+  const handleUnlink = () => {
+    const selection = window.getSelection();
+    if (!selection || !selection.rangeCount) return;
+
+    const range = selection.getRangeAt(0);
+    const container = range.startContainer.parentNode as HTMLElement;
+    if (container && container.tagName === "A") {
+      const parent = container.parentNode;
+      while (container.firstChild) {
+        if (parent) {
+          parent.insertBefore(container.firstChild, container);
+        }
+      }
+      if (parent) {
+        parent.removeChild(container);
+      }
+    }
+    saveState();
+  };
+
+  const handleLineBreak = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      const selection = window.getSelection();
+      if (!selection) return;
+
+      const range = selection.getRangeAt(0);
+      const container = range.startContainer.parentNode as HTMLElement;
+
+      if (container && container.tagName === "LI") {
+        const listNode = container.parentNode as HTMLElement;
+        if (listNode) {
+          const newList = document.createElement(listNode.tagName);
+          const newListItem = document.createElement("li");
+
+          newListItem.appendChild(range.extractContents());
+          newList.appendChild(newListItem);
+          if (listNode.parentNode) {
+            listNode.insertBefore(newListItem, container.nextSibling);
+          }
+
+          range.setStart(newListItem, 0);
+          range.setEnd(newListItem, 0);
+
+          if (
+            container &&
+            container.textContent &&
+            container.textContent.trim() === ""
+          ) {
+            if (listNode.parentNode) {
+              listNode.removeChild(container);
+            }
+          }
+        }
+      } else {
+        if (isUnorderedListActive || isOrderedListActive) {
+          document.execCommand("insertHTML", false, "<br><br>");
+        } else {
+          document.execCommand("insertLineBreak");
+        }
+      }
+
+      saveState();
+    }
+  };
+
+  const updatePreview = () => {
+    const turndownService = new TurndownService();
+    const htmlContent = editorRef.current?.innerHTML || "";
+    const markdownText = turndownService.turndown(htmlContent);
+    setMarkdown(markdownText);
+    setHtmlCode(htmlContent);
```

Comment:
Currently, we are styling in HTML, converting it into markdown and then again we are converting it back into HTML. Why not directly edit markdown? Look into the feasibility of doing this.

Reply:
hmm.. if there is no requirement of live preview (for example showing stars * *) for bold while editing  similar to github comments , then i will go for directly editing the markdown and preview as html